streams/src/main/java/org/apache/kafka/streams/TopologyDescription.java

```diff
@@ -113,8 +115,22 @@
         /**
          * The topic names this source node is reading from.
          * @return comma separated list of topic names or pattern (as String)
+         * @deprecated since 2.1.0. Will be removed in a future major release.
```

Comment:
No need to mention the version -- it's also not clear how long we keep it -- it's at least up to the next major release -- but might be longer. I would rather point to the new method to be use:

`@deprecated use {@link #topicSet()} or {@link #topicPattern()} instead`

This explains users how to rewrite their code and is more helpful.

Reply:
Personally, I don't mind stating the version it was deprecated in; I think this is a common practice.

I agree we shouldn't make claims about when it'll be removed or even say that it will be removed (this is implied by the deprecation itself).

Also +1 for a statement about what to use instead.

Reply:
Why would anyone care about when it was deprecated? And because there is no strict policy about when we remove it, we cannot really claim it in the JavaDocs either?